pages/product-details.js

```diff
@@ -1,338 +1,525 @@
-        <Menu.Item><ItemLink><span>{"New bid"}</span></ItemLink></Menu.Item>
-        <Menu.Item><ItemLink><span>{"View on OpenSea"}</span></ItemLink></Menu.Item>
-        <Menu.Item><ItemLink><span>{"Share"}</span></ItemLink></Menu.Item>
-        <Menu.Item><ItemLink><span>{"Report"}</span></ItemLink></Menu.Item>
-    </DropdownMenu>
-  );
-function ProductPage(){
-   const item = {
-        image:"/images/p1.jpeg",
-        name:"CoinBae #1",
-        favorite:62,
-        category:'🌈 Art',
-        description:`There are only 3. 1 of them in Saltbae's collection, 2 of them are mine. If you buy it, you will have the PSD file. You can edit it as you wish and be the creator of the project. This rare work is a project I've been working on since...`,
-        owner:{
-            avatar:'/images/profpics/1.jpg',
-            name:'Saltbae Nusret'
-        }
-    };
-    return (<>
-        <Header/>
-        <Wrapper>
-            <Content className={`d-sm-flex`}>
-                <ItemImageContainer className="float-none float-sm-left">
-                    <img className={"itemImage"} src={item.image}/>
-                </ItemImageContainer>
-                <ItemInfo className={'float-none float-sm-left'}>
-                <ItemDetails>
-                <ItemDetailsHeader>
-                    <ItemName>{item.name}</ItemName>
-                    <ItemTopButtonContainer>
-                    <button style={{flex:"none"}}><svg viewBox="0 0 17 16" fill="none" width="16" height="16" xlmns="http://www.w3.org/2000/svg"><path d="M8.2112 14L12.1056 9.69231L14.1853 7.39185C15.2497 6.21455 15.3683 4.46116 14.4723 3.15121V3.15121C13.3207 1.46757 10.9637 1.15351 9.41139 2.47685L8.2112 3.5L6.95566 2.42966C5.40738 1.10976 3.06841 1.3603 1.83482 2.97819V2.97819C0.777858 4.36443 0.885104 6.31329 2.08779 7.57518L8.2112 14Z" stroke="rgba(4, 4, 5, 1)" stroke-width="2"></path></svg> 
-                        {" "+ item.favorite}
-                    </button>
-                    <button style={{flex:"none",padding:"0px !important",width:"35px"}}
+  <DropdownMenu className={"mt-3"}>
+    <Menu.Item>
+      <ItemLink>
+        <span>{"New bid"}</span>
+      </ItemLink>
+    </Menu.Item>
+    <Menu.Item>
+      <ItemLink>
+        <span>{"View on OpenSea"}</span>
+      </ItemLink>
+    </Menu.Item>
+    <Menu.Item>
+      <ItemLink>
+        <span>{"Share"}</span>
+      </ItemLink>
+    </Menu.Item>
+    <Menu.Item>
+      <ItemLink>
+        <span>{"Report"}</span>
+      </ItemLink>
+    </Menu.Item>
+  </DropdownMenu>
+);
+function ProductPage() {
+  const [selectedAsset, setSelectedAsset] = useState();
+
+  useEffect(() => {
+    loadAsset(
+      "0x495f947276749ce646f68ac8c248420045cb7b5e",
+      "9870243454359818837048173191911104292730407279767673233444380281837352648705"
+    );
+  }, []);
+
+  const loadAsset = async (tokenAddress, tokenId) => {
+    const asset = await OpenSeaAPI.getAssetDetails(tokenAddress, tokenId);
```

Comment:
i think you can call this function directly in useEffect, no need to create new function as it just  calling another function